src/components/search.js

```diff
@@ -0,0 +1,114 @@
+/* displays Searchbar and enables user to type and filter the output */
+
+import React, { useState } from "react";
+import { useDispatch } from "react-redux";
+import { searchSubreddits } from "../features/subReddits/subRedditsSlice";
+import { debounce } from "lodash";
+import { styled } from "@mui/material/styles";
+import { InputBase, Card } from "@mui/material";
+import SearchIcon from "@mui/icons-material/Search";
+import CancelOutlinedIcon from "@mui/icons-material/CancelOutlined";
+import SubredditsList from "../features/subReddits/subredditsList";
+import { useWindowSize } from "./helperFunctions";
+const ParentDiv = styled("div")(({ theme }) => ({
+  position: "absolute",
+  top: 0,
+  [theme.breakpoints.up("sm")]: {
+    left: "33%",
+    width: "33%",
+  },
+  [theme.breakpoints.down("sm")]: {
+    right: "1rem",
+  },
+}));
+
+const SearchDiv = styled("div")(({ theme }) => ({
+  borderRadius: 50,
+  backgroundColor: "#fafafa",
+  color: "rgba(0, 0, 0, 0.54)",
+  display: "flex",
+  alignItems: "center",
+  margin: "4.5px auto",
+  width: "100%",
+}));
+
+const StyledInputBase = styled(InputBase)(({ theme }) => ({
+  color: "black",
+  "& .MuiInputBase-input": {
+    padding: theme.spacing(1, 1, 1, 0),
```

Comment:
very good, anything related to margin or padding should be like that, it's more maintainable